core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/logprovider/provider.go

```diff
@@ -538,13 +503,7 @@ func (p *logEventProvider) readLogs(ctx context.Context, latest int64, filters [
 
 func (p *logEventProvider) syncBufferFilters() error {
 	p.lock.RLock()
-	buffVersion := p.opts.BufferVersion
-	p.lock.RUnlock()
+	defer p.lock.RUnlock()
```

Comment:
we can probably remove the Lock/unlock completely here since we dont need to read buffer version